File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.java
class Solution {
    public int search(int[] nums, int target) {
        int n = nums.length;
        int left = 0; int right = n - 1; 

        while (left < right)
        {   
            int mid = (left + right)/2; 
            if (nums[mid] < nums[right]) // 1 2 3 4 5 
                right = mid; 
            else // mid > right 
                left = mid + 1; // 3 4 5 1 2 
        }

        System.out.println(left); 
        int x1 = binarySearch(nums, 0, left, target); 
        if (x1 != -1)
            return x1; 
        return binarySearch(nums, left, nums.length - 1, target); 
    }

    private int binarySearch(int[] nums, int left, int right, int target)
    {
        while (left <= right)
        {
            int mid = (left + right)/2; 
            if (nums[mid] == target)
                return mid;
            else if (nums[mid] < target)
                left = mid + 1; 
            else 
                right = mid - 1; 
        }

        return -1; 
    }
}

File: Data Structures & Algorithms/binary-search/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1
        while l <= r: 
            mid = l + (r - l) // 2
            num = nums[mid]
            if target == num:
                return mid 
            elif target > num: 
                l = mid + 1 
            else: 
                r = mid - 1

        return -1 